tests/analog/test_patterns.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
+
+import pytest
+import torch
+from metrics import ATOL_DICT
+
+from qadence import (
+    AnalogRX,
+    AnalogRY,
+    BackendName,
+    DiffMode,
+    Parameter,
+    QuantumCircuit,
+    QuantumModel,
+    Register,
+    chain,
+    total_magnetization,
+)
+from qadence.analog.addressing import AddressingPattern
+from qadence.analog.interaction import add_interaction
+from qadence.backends.pulser.config import Configuration
+
+
+@pytest.mark.parametrize(
+    "amp,det",
+    [(0.0, 10.0), (15.0, 0.0), (15.0, 9.0)],
+)
+@pytest.mark.parametrize(
+    "spacing",
+    [8.0, 30.0],
+)
+def test_pulser_pyq_addressing(amp: float, det: float, spacing: float) -> None:
+    n_qubits = 3
+    x = Parameter("x")
+    block = chain(AnalogRX(3 * x), AnalogRY(0.5 * x))
+    reg = Register(support=n_qubits, spacing=spacing)
+    circ = QuantumCircuit(reg, block)
+
+    # define addressing patterns
+    rand_weights_amp = torch.rand(n_qubits)
+    rand_weights_amp = rand_weights_amp / rand_weights_amp.sum()
+    w_amp = {i: rand_weights_amp[i] for i in range(n_qubits)}
+    rand_weights_det = torch.rand(n_qubits)
+    rand_weights_det = rand_weights_det / rand_weights_det.sum()
+    w_det = {i: rand_weights_det[i] for i in range(n_qubits)}
+    p = AddressingPattern(
+        n_qubits=n_qubits,
+        det=det,
+        amp=amp,
+        weights_det=w_det,
+        weights_amp=w_amp,
+    )
+
+    values = {"x": torch.linspace(0.5, 2 * torch.pi, 50)}
+    obs = total_magnetization(n_qubits)
+    conf = Configuration(addressing_pattern=p)
+
+    # define pulser backend
+    model = QuantumModel(
+        circuit=circ,
+        observable=obs,
+        backend=BackendName.PULSER,
+        diff_mode=DiffMode.GPSR,
+        configuration=conf,
+    )
+    expval_pulser = model.expectation(values=values)
+
+    # define pyq backend
+    int_circ = add_interaction(circ, pattern=p)
+    model = QuantumModel(
+        circuit=int_circ, observable=obs, backend=BackendName.PYQTORCH, diff_mode=DiffMode.AD
+    )
+    expval_pyq = model.expectation(values=values)
+
+    torch.allclose(expval_pulser, expval_pyq, atol=ATOL_DICT[BackendName.PULSER])
+
+
+@pytest.mark.flaky(max_runs=10)
+def test_addressing_training() -> None:
+    n_qubits = 3
+    reg = Register(support=n_qubits, spacing=8)
+    f_value = torch.rand(1)
+
+    # define training parameters
+    w_amp = {i: Parameter(f"w_amp{i}", trainable=True) for i in range(n_qubits)}
+    w_det = {i: Parameter(f"w_det{i}", trainable=True) for i in range(n_qubits)}
```

Comment:
Maybe we should allow direct string input here and have those converted to `VariationalParameter` automatically to be consistent with other parameter inputs in Qadence.

Reply:
Added this possibility